### PR TITLE
[BEAM-3339] Create a post-release test of the nightly snapshots

### PR DESCRIPTION
**Pull request by alanmyrvold**
Follow this checklist to help us incorporate your contribution quickly and easily:

 - [ ] Make sure there is a [JIRA issue](https://issues.apache.org/jira/projects/BEAM/issues/) filed for the change (usually before you start working on it).  Trivial changes like typos do not require a JIRA issue.  Your pull request should address just this issue, without pulling in other changes.
 - [ ] Each commit in the pull request should have a meaningful subject line and body.
 - [ ] Format the pull request title like `[BEAM-XXX] Fixes bug in ApproximateQuantiles`, where you replace `BEAM-XXX` with the appropriate JIRA issue.
 - [ ] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
 - [ ] Run `mvn clean verify` to make sure basic checks pass. A more thorough check will be performed on your pull request automatically.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

---

### CODE REVIEW

**Comment by alanmyrvold** [comment]
Run Python PreCommit

**Comment by alanmyrvold** [comment]
Parameterized the Jenkins job with the label and url with the defaults being the nightly build, which should allow running against a staged artifact too.

**Comment by alanmyrvold** [comment]
Run Java Gradle PreCommit
Run Go PreCommit

**Comment by alanmyrvold** [comment]
Parameterized the Jenkins job ... @lukecwik can you take a look?

**Comment by alanmyrvold** [comment]
Run Go PreCommit

**Comment by alanmyrvold** [comment]
Run Java Gradle PreCommit

File: release/TestHelper.groovy [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+   // Both documents the overal scenario and creates a clean temp directory
+   def describe(String desc) {
+     var.startDir = File.createTempDir()
+     var.startDir.deleteOnExit()
+     var.curDir = var.startDir
+     print "*****\n* Scenario: ${desc}\n*****\n"
+   }
+   
+   // Just document the intention of a set of steps
+   def it(String desc) {
```
Comment (lukecwik) [inline release/TestHelper.groovy]: rename `it` -> `intent`

File: release/TestHelper.groovy [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/*
+ * Helper functions to make writing a test similar to the quickstart
+ * instructions from https://beam.apache.org/get-started/quickstart-java/
+ */
+class TestHelper {
```
Comment (lukecwik) [inline release/TestHelper.groovy]: Rename `TestHelper` -> `TestScripts`

File: release/TestHelper.groovy [hunk 3]
```diff
@@ -0,0 +1,140 @@
...
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/*
+ * Helper functions to make writing a test similar to the quickstart
```
Comment (lukecwik) [inline release/TestHelper.groovy]: `Helper functions` -> `Scripting functions`

**Comment by lukecwik** [comment]
Thanks for the changes.

**Pull request merged**